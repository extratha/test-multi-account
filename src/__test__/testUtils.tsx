import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
