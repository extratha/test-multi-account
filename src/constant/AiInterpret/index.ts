export const FIELD_NAME = {
  GENDER: "gender",
  STOOL_COLOR: "stool_color",
  STOOL_WBC: "stool_wbc",
  STOOL_RBC: "stool_rbc",
  STOOL_PARASITE: "stool_parasite",
  STOOL_OCCULT_BLOOD: "stool_occult_blood",
  STOOL_CULTURE: "stool_culture",
  URINE_COLOR: "urine_color_value",
  URINE_SP_GR_VALUE: "urine_sp_gr_value",
  URINE_PH_VALUE: "urine_ph_value",
  URINE_WBC_VALUE: "urine_wbc_value",
  URINE_PROTEIN_VALUE: "urine_protein_value",
  URINE_GLUCOSE_VALUE: "urine_glucose_value",
  URINE_ERYTHROCYTE_VALUE: "urine_erythrocyte_value",
  URINE_RBC_VALUE: "urine_rbc_value",
};
