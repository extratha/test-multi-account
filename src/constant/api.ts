
export const API = {
  PATH: {
    login: 'auth/login',
    setNewPassword: 'auth/reset-password',
    changePassword: 'auth/change-password',
    forgotPassword: 'auth/forgot-password',
    exampleDataList: 'lab/examples',
  },
  HTTP_METHODS: {
    GET: 'GET',
    POST: 'POST',
    PUT: 'PUT',
    PATCH: 'PATCH',
    DELETE: 'DELETE',
  },
  ERROR: {
    UNAUTHENTICATION: {
      MESSAGE: 'unauthenticated',
      CODE: 401,
    },
    login: {
      INVALID_CREDENTIAL: {
        MESSAGE: 'invalid email or password',
        CODE: 400,
      },
    }
  },
};
