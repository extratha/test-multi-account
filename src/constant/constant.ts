export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};
