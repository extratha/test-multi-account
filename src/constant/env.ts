export const staticEnvConfig = {
  apiUrl:
    process.env.NEXT_PUBLIC_API_BASE_PATH ?? 'MOCK_NEXT_PUBLIC_API_BASE_PATH',
  domainUrl:
    process.env.NEXT_PUBLIC_BASE_DOMAIN_URL ??
    'MOCK_NEXT_PUBLIC_BASE_DOMAIN_URL',
};
