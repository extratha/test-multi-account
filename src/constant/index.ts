export const ENV = {
  BASE_API_URL: process.env.NEXT_PUBLIC_BASE_API_URL || "MOCK_NEXT_PUBLIC_BASE_API_URL",
  FIREBASE_API_KEY: process.env.NEXT_PUBLIC_FIREBASE_API_KEY || "MOCK_NEXT_PUBLIC_FIREBASE_API_KEY",
  FIREBASE_AUTH_DOMAIN: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN || "MOCK_NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN",
  FIREBASE_PROJECT_ID: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_PROJECT_ID",
  FIREBASE_STORAGE_BUCKET:
    process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET || "MOCK_NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET",
  FIREBASE_MESSAGING_SENDER_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID",
  FIREBASE_APP_ID: process.env.NEXT_PUBLIC_FIREBASE_APP_ID || "MOCK_NEXT_PUBLIC_FIREBASE_APP_ID",
  FIREBASE_MEASUREMENT_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID",
};

export const SESSION = {
  ACCESS_TOKEN: "access-token",
  REFRESH_TOKEN: "refresh-token",
  RESET_PASSWORD_TOKEN: "reset-password-token",
  X_ROBOT_TOKEN: "x-robot-token",
};

export const MENU_CONDITION = {
  LOGOUT: "logout",
};

export const NAVIGATION = {
  NOT_FOUND: "/not-found",
  LOGIN: "/login",
  HOME: "/home",
  FORGET_PASSWORD: "/forget-password",
  SET_NEW_PASSWORD: "/set-new-password",
  CONSENT_TERMS_CONDITIONS: "/consents/terms-conditions",
  CONSENT_PRIVACY_POLICIES: "/consents/privacy-policies",
  AI_INTERPRET_RESULT: "/ai-interpret/interpret",
  AI_INTERPRET_TRY_EXAMPLE_DATA: "/ai-interpret/try-example-data",
  AI_INTERPRET_TRY_INPUT_DATA: "/ai-interpret/try-input-data",
  SYMPTOM_CHECKER: "/symptom-checker",
  SETTINGS_TERMS_AND_CONDITIONS: "/settings/terms-and-conditions",
  SETTINGS_PRIVACY_POLICY: "/settings/privacy-policy",
  PUBLIC_TERMS_AND_CONDITIONS: "/public/consents/terms-conditions",
  PUBLIC_PRIVACY_POLICY: "/public/consents/privacy-policies",
};

export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};

export const GROUP_NAME = {
  GENERAL_CHECK_UP: "generalCheckUp",
  HEMATOLOGY_BG: "hematologyBG",
  HEMATOLOGY_CBC: "hematologyCBC",
  LIPID_PROFILE: "lipidProfile",
  BLOOD_CHEMISTRY: "bloodChemistry",
  RENAL_FUNCTION_TEST: "renalFunctionTest",
  LIVER_FUNCTION_TEST: "liverFunctionTest",
  URINE_ANALYSIS: "urineAnalysis",
  STOOL_EXAMINATION: "stoolExamination",
  PAP_SMEAR: "papSmear",
  TUMOR_MARKER: "tumorMarker",
};

export const GENERAL_CHECK_UP = {
  AGE: "age",
  GENDER: "gender",
  BODY_WEIGHT: "body_weight",
  BODY_HEIGHT: "body_height",
  WAIST_LINE: "waist_line",
  BLOOD_PRESSURE_SYSTOLIC: "blood_pressure_systolic",
  BLOOD_PRESSURE_DIASTOLIC: "blood_pressure_diastolic",
  PULSE: "pulse",
  RESPIRATORY_RATE: "respiratory_rate",
  TEMPERATURE: "temperature",
};

export const HEMATOLOGY_BLOOD = {
  BLOOD_GROUP: "blood_group",
  BLOOD_GROUP_RH: "blood_group_rh",
};

export const HEMATOLOGY_BLOOD_GROUP = {
  HEMATOLOGY_BLOOD: "bloodGroup",
};

export const HEMATOLOGY_CBC = {
  HB: "hb_value",
  HCT: "hct_value",
  RBC_MORPHOLOGY: "rbc_morphology_value",
  MCV: "mcv_value",
  MCH: "mch_value",
  MCHC: "mchc_value",
  RDW: "rdw_value",
  WBC: "wbc_value",
  NEUTROPHIL: "neutrophil_value",
  LYMPHOCYTE: "lymphocyte_value",
  MONOCYTE: "monocyte_value",
  EOSINOPHIL: "eosinophil_value",
  BASOPHIL: "basophil_value",
  PLATELET_COUNT: "platelet_count_value",
};

export const HEMATOLOGY_CBC_GROUP = {
  HEMOGLOBIN: "hemoglobin",
  WHITE_BLOOD: "whiteBlood",
  RED_BLOOD: "redBlood",
};

export const INTERPRET_STATUS = {
  PENDING: "pending",
  SUCCESS: "success",
  FAILED: "failed",
};

export const ASR_SERVICE = {
  ORDER_ASR: "order-asr",
  SOAP_ASR: "soap-asr",
};

export const ASR_SERVICE_KEY: Record<string, string> = {
  [ASR_SERVICE.ORDER_ASR]: "orderAsr",
  [ASR_SERVICE.SOAP_ASR]: "soapAsr",
};

export const CONSENT_TYPE = {
  TERMS_CONDITIONS: "terms-conditions",
  PRIVACY_POLICIES: "privacy-policies",
};

export const SUBMIT_CONSENT_TYPE = {
  [CONSENT_TYPE.TERMS_CONDITIONS]: "terms",
  [CONSENT_TYPE.PRIVACY_POLICIES]: "privacy_policies",
};
