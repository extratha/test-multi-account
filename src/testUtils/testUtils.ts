import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
