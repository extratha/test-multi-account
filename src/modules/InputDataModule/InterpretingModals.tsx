import { IconErrorCircle, ImageLoadingStack } from "@/assets";
import { INTERPRET_STATUS } from "@/constant/constant";
import useTranslation from "@/locales/useLocale";
import { BaseModalProps } from "@/store";
import { Button, Stack, styled, Typography } from "@mui/material";
import Image from "next/image";

const ModalStackWrapper = styled(Stack)(({ theme }) => ({
  alignItems: "center",
  color: theme.palette.background.paper,
}));

const StatusImageInModal = styled(Image)({
  width: "100px",
  height: "100px",
});

const StatusMessage = styled(Typography)({
  marginTop: "18px",
  whiteSpace: "pre-line",
});

const ButtonInModal = styled(Button)(({ theme }) => ({
  width: "150px",
  height: "46px",
  marginTop: "36px",
  backgroundColor: "transparent",
  color: theme.palette.background.paper,
  border: `2px solid ${theme.palette.background.paper}`,
  fontWeight: 600,
}));

export interface InterpretingModalsProps extends BaseModalProps {
  interpretStatus: string;
}

const InterpretingModals = (props: InterpretingModalsProps) => {
  const { closeModal, interpretStatus } = props;
  const { translation } = useTranslation();

  return (
    <ModalStackWrapper>
      {interpretStatus === INTERPRET_STATUS.PENDING && (
        <>
          <StatusImageInModal data-testid="interpret-image-pending" alt="" src={ImageLoadingStack} />
          <StatusMessage variant="bodyBold">{translation("AiInterpret.text.interpreting")}</StatusMessage>
          <ButtonInModal data-testid="modal-ok-button" onClick={closeModal}>
            {translation("AiInterpret.button.ok")}
          </ButtonInModal>
        </>
      )}
      {interpretStatus === INTERPRET_STATUS.FAILED && (
        <>
          <IconErrorCircle data-testid="interpret-image-failed" />
          <StatusMessage variant="bodyBold" textAlign="center">
            {translation("AiInterpret.text.failedInterpret")}
          </StatusMessage>
          <ButtonInModal data-testid="modal-retry-button" onClick={closeModal}>
            {translation("AiInterpret.button.retry")}
          </ButtonInModal>
        </>
      )}
    </ModalStackWrapper>
  );
};

export default InterpretingModals;
