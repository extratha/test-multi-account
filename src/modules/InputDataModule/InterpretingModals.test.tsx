import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { INTERPRET_STATUS } from "@/constant/constant";
import { render } from "@/testUtils/testUtils";
import InterpretingModals, { InterpretingModalsProps } from "./InterpretingModals";

describe("InterpretingModals", () => {
  let props: InterpretingModalsProps;

  beforeEach(() => {
    props = {
      interpretStatus: INTERPRET_STATUS.PENDING,
      closeModal: jest.fn(),
    };
  });

  const renderInterpretingModals = () => {
    return render(<InterpretingModals {...props} />);
  };

  it("should render correctly", () => {
    const { asFragment } = renderInterpretingModals();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display InterpretingModals with status failed", async () => {
    props.interpretStatus = INTERPRET_STATUS.FAILED;
    renderInterpretingModals();

    expect(screen.getByTestId("interpret-image-failed")).toBeInTheDocument();
  });

  it("should call closeModal prop when click retry button", async () => {
    props.interpretStatus = INTERPRET_STATUS.PENDING;
    renderInterpretingModals();

    await userEvent.click(screen.getByTestId("modal-ok-button"));

    expect(props.closeModal).toHaveBeenCalled();

    props.interpretStatus = INTERPRET_STATUS.FAILED;
    renderInterpretingModals();

    await userEvent.click(screen.getByTestId("modal-retry-button"));

    expect(props.closeModal).toHaveBeenCalled();
  });
});
