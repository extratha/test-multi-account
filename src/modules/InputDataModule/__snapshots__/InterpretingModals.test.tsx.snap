// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterpretingModals should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
}

.emotion-1 {
  width: 100px;
  height: 100px;
}

.emotion-2 {
  margin: 0;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.5;
  margin-top: 18px;
  white-space: pre-line;
}

.emotion-3 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 2px solid #fff;
  margin: 0;
  border-radius: 4px;
  padding: 6px 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  font-family: fontFamily;
  font-weight: 600;
  font-size: 12px;
  line-height: 1.5;
  text-transform: uppercase;
  min-width: 64px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  width: 150px;
  height: 46px;
  margin-top: 36px;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(33, 150, 243, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiStack-root emotion-0"
  >
    <img
      alt=""
      class="emotion-1"
      data-testid="interpret-image-pending"
      src="/img.jpg"
    />
    <p
      class="MuiTypography-root MuiTypography-bodyBold emotion-2"
    >
      กำลังประมวลผล
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary emotion-3"
      data-testid="modal-ok-button"
      tabindex="0"
      type="button"
    >
      ยกเลิก
      <span
        class="MuiTouchRipple-root emotion-4"
      />
    </button>
  </div>
</DocumentFragment>
`;
